modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexRenamedSettingTests.java

```diff
@@ -8,13 +8,16 @@
 
 package org.opensearch.index.reindex;
 
+import org.hamcrest.MatcherAssert;
 import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.test.OpenSearchTestCase;
 
 import java.util.Arrays;
 import java.util.List;
 
+import static org.hamcrest.Matchers.equalTo;
```

Comment:
According to suggestion from @andrross , I updated all the assertions in this file to use APIs from Hamcrest. I will update the PR description later.
although `assertThat()` is also available in junit, but it's deprecated. (https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertThat(java.lang.String,%20T,%20org.hamcrest.Matcher))